src/Core/CoreMacros.hpp

```diff
@@ -246,58 +254,78 @@ typedef double Scalar;
 // and the debugger will break in assert.c or and not where the calling code
 // uses assert(). CORE_ASSERT guarantees a breakpoint when in a debugger,
 // and always prints a useful message.
 // CORE_WARN_IF has the same effect but it will only print a message.
-#ifdef CORE_DEBUG
-#define CORE_ASSERT( EXP, DESC )                       \
-    MACRO_START                                        \
-    if (UNLIKELY(!(EXP))) {                            \
-        fprintf(stderr,                                \
-                "%s:%i: Assertion `%s` failed : %s\n", \
-                __FILE__,__LINE__, #EXP, DESC);        \
-        BREAKPOINT(0);                                 \
-    } else {}                                          \
+#ifdef CORE_ENABLE_ASSERT
+#define CORE_ASSERT( EXP, DESC )    \
+    MACRO_START                     \
+    if (UNLIKELY(!(EXP))) {         \
+        REPORT_FAIL(EXP, DESC, "%s:%i: Assertion `%s` failed : %s\n");\
+        BREAKPOINT(0);              \
+    } else {}                       \
     MACRO_END
 
-#define CORE_WARN_IF( EXP, DESC )                      \
-    MACRO_START                                        \
-    if (UNLIKELY((EXP))) {                             \
-        fprintf(stderr,                                \
-                "%s:%i: WARNING `%s` : %s\n",          \
-                __FILE__,__LINE__, #EXP, DESC);        \
-    } else{}                                           \
+#define CORE_WARN_IF( EXP, DESC )  \
+    MACRO_START                    \
+    if (UNLIKELY((EXP))) {         \
+        REPORT_FAIL(EXP, DESC, "%s:%i: WARNING `%s` : %s\n");\
+    } else{}                       \
     MACRO_END
 #else
 #define CORE_ASSERT( EXP, DESC ) // nothing
 #define CORE_WARN_IF( EXP, DESC ) // nothing
 #endif
 
 // Print an error and break, even in release.
-#define CORE_ERROR( DESC )                  \
-    MACRO_START                             \
-    fprintf(stderr,                         \
-            "%s:%i: ERROR : %s\n",          \
-            __FILE__,__LINE__, DESC);       \
-    BREAKPOINT(0);                          \
-    exit(EXIT_FAILURE);                     \
+#define CORE_ERROR( DESC )                     \
+    MACRO_START                                \
+    REPORT_FAIL(ERROR, DESC, "%s:%i %s: %s\n");\
+    BREAKPOINT(0);                             \
+    exit(EXIT_FAILURE);                        \
     MACRO_END
 
 // Print an error and break if condition is not met, even in release
-#define CORE_ERROR_IF( EXP, DESC )          \
-    MACRO_START                             \
-    if( UNLIKELY(!(EXP)) ) {                \
-        CORE_ERROR( DESC );                 \
-    }                                       \
+#define CORE_ERROR_IF( EXP, DESC ) \
+    MACRO_START                    \
+    if( UNLIKELY(!(EXP)) ) {       \
+        CORE_ERROR( DESC );        \
+    }                              \

```

Comment:
Actually this could directly call REPORT_FAIL with the correct expression. Fixing that soon